### DISCUSSION TITLE
[Bug Report] 4.0.0-rc.6 宫格组件打包后没有边框线

### DISCUSSION THREAD

**Discussion by tlerbao** [youzan/vant#11192] [General]
### 重现链接

无法提供重现连接看截图

### Vant 版本

4.0.0-rc.6

### 描述一下你遇到的问题。

宫格组件本地run dev没问题，打包上线后 没有线条如下截图
**打包后**
<img width="410" alt="image" src="https://user-images.githubusercontent.com/20442232/198232620-0bf90d05-f3eb-4810-bee0-5262ede3b128.png">

**本地预览**
<img width="414" alt="image" src="https://user-images.githubusercontent.com/20442232/198232709-f4dad380-2b96-477e-b8f5-4c364d3b6577.png">

### 重现步骤

宫格

### 设备/浏览器

chrome

**Comment by wjw-gavin** [comment]
`4.0.0-rc.6`版本，我这打包没复现你这种情况...

![image](https://user-images.githubusercontent.com/19986739/198261758-762051dc-e4c0-4ab6-adc5-6c8c509d782d.png)

**Comment by tlerbao** [comment]
@wjw-gavin 但是我的为何不行不知道 ,如果github能私信你线上地址就好了，不想在这里公布

```
<van-grid clickable :column-num="2">
      <van-grid-item icon="friends" text="客户录入排行" class="grid-item" to="/crm/report/cus/total" />
      <van-grid-item icon="shop" text="客户民族占比" to="/crm/report/cus/ethnic" badge="New" />
      <van-grid-item icon="service" text="客户年龄占比" to="/crm/report/cus/age" badge="New" />
      <van-grid-item icon="goods-collect" text="签单总数排行" to="/crm/report/order/total" />
      <van-grid-item icon="cluster" text="最大单值排行" to="/crm/report/order/max" />
      <van-grid-item icon="font" text="平均单值排行" to="/crm/report/order/avg" />
      <van-grid-item icon="eye" text="订单渠道来源" to="/crm/report/order/source" dot />
      <van-grid-item icon="expand" text="订单小区分布" to="/crm/report/order/community" badge="99+" />
    </van-grid>
```

**Comment by wjw-gavin** [comment]
![image](https://user-images.githubusercontent.com/19986739/198491147-d1bd3149-65dc-4aee-a114-53974f71ffdd.png)

我用的你的代码，打包后也是正常。

检查一下你的css，看是否被别的影响了？

**Comment by tlerbao** [comment]
@wjw-gavin 没有引入任何其他css啊
<img width="1207" alt="image" src="https://user-images.githubusercontent.com/20442232/198496399-22225e98-087d-4e6b-b254-463e7dfde60d.png">

**Comment by wjw-gavin** [comment]
我仔细看了一下，你的这个是有边框的，只是颜色太浅了...我是放大图片仔细看才看到的...😂
你改下 `border: 0 solid #000;` 试试

**Comment by tlerbao** [comment]
这有个git不知道你能不能看清楚，取消勾选 border: 0 solid var(--van-border-color); 边框就会显示出来，勾选就又没了。


![iShot_2022-10-28_11 48 51](https://user-images.githubusercontent.com/20442232/198498474-0ea203e9-7887-4ccb-badd-51f2cc22fdfd.gif)

**Comment by wjw-gavin** [comment]
要不你提供个最小仓库，不然我也没法复现你这情况...

**Comment by tlerbao** [comment]
@wjw-gavin 我github邀请你了 我 QQ/wx 327358

**Comment by wjw-gavin** [comment]
我也没找到问题在哪，发现不单单是grid没有边框，只要是用到细边框的地方，都没有。（但是总感觉跟你用的那些auto-import有关...）o(╥﹏╥)o

**Comment by tlerbao** [comment]
@wjw-gavin 你说跟auto-import 有关吧，把同样的模板内容放到/index.vue打包就有边框线，放单/crm/report/index.vue下就没有了，我就很是奇怪，太灵异了哈哈。

**Comment by wjw-gavin** [comment]
> @wjw-gavin 你说跟auto-import 有关吧，把同样的模板内容放到/index.vue打包就有边框线，放单/crm/report/index.vue下就没有了，我就很是奇怪，太灵异了哈哈。

是的，我试了。而且比如button、cell等，只要存在0.5（细）边框的都不显示...

**Reply by wjw-gavin** [reply]
发现 ”把同样的模板内容放到/index.vue打包就有边框线“，这种情况打包的css文件有一个，原始情况打包的css文件有三个。因为我没用过这些，具体也是不清楚...

**Comment by wjw-gavin** [comment]
感觉应该不是组件本身的问题，我先移动到讨论区啦，有进展咱在那里讨论😂

**Reply by tlerbao** [reply]
刚又在给你那个测试仓库的基础上先后卸载了tailwindcss postcss autoprefixer unplugin-auto-import vite-plugin-vue-layouts，还是不行，真的是奇怪了，也没有其他css样式，难道是打包css引入顺序？
我菜，哈哈，如果大神们不帮忙看看，我这边估计很难有进展了，已经很努力在测试了，排除法哈哈。

**Reply by tlerbao** [reply]
@wjw-gavin 测试到今天了哈，我又新init了一个新的项目，然后一个一个装插件，把项目文件复制过来，最后插件都一样了，新init的项目有边框了，我对比发现，出问题的项目和新项目就是打包css加载顺序不同导致的 .van-grid-item__content:after 在前 [class*=van-hairline]:after在后 就会有边框线，反之 有问题的是[class*=van-hairline]:after 在前 
<img width="580" alt="image" src="https://user-images.githubusercontent.com/20442232/198819652-7a6cc40b-0f7c-4d83-9219-38c5f1eac53e.png">

似乎现在两个项目配置都完全一样了，但是为何打包的css加载顺序不一样。。。愁死了。

**Reply by chenjiahan** [reply]
vite 低版本是有 CSS 加载顺序的 bug 的，可以参考 https://github.com/youzan/vant/issues/11049

**Reply by tlerbao** [reply]
@chenjiahan 果然，升级了vite就好了，折腾两天。。。。

**Reply by chenjiahan** [reply]
😂
